optimade/filterparser/lark_parser.py

```diff
@@ -30,13 +30,13 @@ def get_versions() -> Dict[Tuple[int, int, int], Dict[str, str]]:
         A mapping from version, variant to grammar file name.
 
     """
-    dct = defaultdict(dict)
+    dct: Dict[Tuple[int, int, int], Dict[str, Path]] = defaultdict(dict)
     for filename in Path(__file__).parent.joinpath("../grammar").glob("*.lark"):
         tags = filename.stem.lstrip("v").split(".")
-        version = tuple(map(int, tags[:3]))
+        version = tuple(map(int, tags[:3]))  # ignore: type[index]
```

Comment:
I think it should be # type: ignore[index]
But since mypy does not give any warning, I think we can remove it.
```suggestion
        version = tuple(map(int, tags[:3]))
```